### PR TITLE
refactor(back-top): refactor backTop with ts

### PR DESCRIPTION
**Pull request by anncwb**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/back-top/backTopProps.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    default: 400,
+  },
+  target: {
+    type: Function as PropType<Fn>,
+  },
+  prefixCls: {
+    type: String as PropType<string>,
+  },
+  onClick: {
+    type: Function as PropType<Fn>,
```
Comment (Amour1688) [inline components/back-top/backTopProps.ts:16]: 这个断言和不断言有啥区别

File: examples/components/BackTop.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import BackTop from '../../components/back-top';
```
Comment (Amour1688) [inline examples/components/BackTop.tsx:1]: 不相关的文件不需要提交

File: components/back-top/backTopProps.ts (hunk 1 above)
Comment (anncwb) [reply components/back-top/backTopProps.ts:16]: 对于单组件，这个可以直接校验props，并不需要独立的类型定义文件，且对template和tsx都可以直接自动提示类型信息

File: examples/components/BackTop.tsx (hunk 2 above)
Comment (anncwb) [reply examples/components/BackTop.tsx:1]: 好的

Comment (anncwb) [reply examples/components/BackTop.tsx:1]: 需要重新提交pr吗，或者后面提交的新的组件这样改。我后面还有几个组件写好了。大概都这这种风格，等你定一下我在提交pr

File: components/back-top/backTopProps.ts (hunk 1 above)
Comment (anncwb) [reply components/back-top/backTopProps.ts:16]: 这个对于工具的自动联想也有挺大帮助

**Comment by anncwb** [comment]
@Amour1688 我后面有提交了一个，没看到你前面发的信息，如果不符合。我关掉重新修改提交pr

**Comment by Amour1688** [comment]
> @Amour1688 我后面有提交了一个，没看到你前面发的信息，如果不符合。我关掉重新修改提交pr

每个 PR，提交一个组件，多个组件分开提。直接在这个 PR 上改就行

**Comment by anncwb** [comment]
> > @Amour1688 我后面有提交了一个，没看到你前面发的信息，如果不符合。我关掉重新修改提交pr
> 
> 每个 PR，提交一个组件，多个组件分开提。直接在这个 PR 上改就行

好的，重新弄下

**Comment by anncwb** [comment]
@Amour1688 backTop已经重新调整。等你合并我在提交下一个

File: components/back-top/backTopTypes.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export interface BackTopState {
+  scrollEvent: {
+    remove: () => void;
+  } | null;
+  visible: boolean;
+}
+
+export interface BackTopProps extends HTMLAttributes {
+  prefixCls?: string;
+  visibilityHeight: number;
```
Comment (xrkffgg) [inline components/back-top/backTopTypes.ts:14]:
```suggestion
  visibilityHeight?: number;
```

File: components/back-top/index.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+} from 'vue';
+import classNames from '../_util/classNames';
+import addEventListener from '../vc-util/Dom/addEventListener';
+import getScroll from '../_util/getScroll';
+import getTransitionProps from '../_util/getTransitionProps';
+import { defaultConfigProvider, ConfigConsumerProps } from '../config-provider';
+import scrollTo from '../_util/scrollTo';
+import { BackTopProps, BackTopState } from './backTopTypes';
+import backTopProps from './backTopProps';
+function getDefaultTarget() {
```
Comment (xrkffgg) [inline components/back-top/index.tsx:19]:
```suggestion

function getDefaultTarget() {
```

**Pull request closed without merging**